### PR TITLE
Add changes in IoTS-1.0.0-M3 to M2

### PR DESCRIPTION
**Pull request by charithag**
Merge changes

### CODE REVIEW

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java [hunk 1]
```diff
@@ -84,18 +80,28 @@ public void initConnector() {
...
-        String deviceId = from.substring(0, indexOfAt);
-        String resource = from.substring(indexOfSlash + 1, from.length());
-
-        if ((inbound_message != null)&&(resource.equals(DroneConstants.MESSAGE_RESOURCE)) ){
-            messageTransformer.messageTranslater(inbound_message);
-        } else {
-            log.error("Message is empty or it is not belongs to "+ DroneConstants.DEVICE_ID);
+        try{
+            String from = message.getFrom();
+            String inbound_message = message.getBody();
```
Comment (kasundananjaya) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java]: Follow proper variable naming conventions!

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java [hunk 2]
```diff
@@ -84,24 +79,23 @@ public void onOpen(Session session){
 
     @OnMessage
     public void onMessage(String message, Session session){
-        try {
             while(true){
-                if((messageController !=null) && (!messageController.isEmptyQueue())){
-                    String message1 = messageController.getMessage();
-                    session.getBasicRemote().sendText(message1);
+                try{
+                    if((messageController !=null) && (!messageController.isEmptyQueue())){
```
Comment (kasundananjaya) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java]: Fix formatting issues.

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java [hunk 3]
```diff
@@ -84,24 +79,23 @@ public void onOpen(Session session){
...
+                        String message1 = messageController.getMessage();
```
Comment (kasundananjaya) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java]: Follow proper variable naming conventions!

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java [hunk 4]
```diff
@@ -84,18 +80,28 @@ public void initConnector() {
...
+            int indexOfSlash = from.indexOf("/");
+            if(indexOfSlash==0){
```
Comment (kasundananjaya) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java]: Fix formatting issues.

Comment (charithag) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java]: Fix with PR https://github.com/wso2/carbon-device-mgt-plugins/pull/206

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java (hunk 1 above)
Comment (charithag) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/transport/DroneAnalyzerXMPPConnector.java]: Ack, This code no longer available in the repository.

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java (hunk 2 above)
Comment (charithag) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java]: Ack, This code is no longer available in the repository. Hence unable to fix.

File: components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java (hunk 3 above)
Comment (charithag) [inline components/device-mgt-iot-droneanalyzer/org.wso2.carbon.device.mgt.iot.droneanalyzer.controller.api/src/main/java/org/wso2/carbon/device/mgt/iot/droneanalyzer/controller/api/impl/DroneRealTimeService.java]: Ack, This code is no longer available in the repository. Hence unable to fix.

**Pull request merged**